### PR TITLE
[full-ci] Resource ids in urls

### PR DESCRIPTION
**Pull request by kulmann**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/6247
- Fixes https://github.com/owncloud/web/issues/6899
- Fixes https://github.com/owncloud/web/issues/7714
- Fixes https://github.com/owncloud/web/issues/7715

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
- [x] add `parentFolderId` in all `buildResource` helpers.
- [x] fix routes in upload related source files (UploadInfo.vue, useUpload.ts, useUploaderHelpers.ts)
- [x] remove code duplication in multiple folder loaders: can be merged into one folder loader.
- [x] use webdav meta request to get file path by file id in folder loader (if loading by path fails) and try again
- [x] rewrite route to correct path if invalid
- [x] appDefaults: use fileId in FileContext if present
- [x] changelog items
- [x] repeated click on nav items doesn't re-resolve the correct space (for personal/all files and for trash), only for oc10.

## Followup tasks / out of scope for this PR
- [ ] breadcrumbs are currently built from parent paths segments, we'd need a propfind to each parent folder to build routes with fileIds (if we'd switched to propfinds for gathering share type information, we'd already have that... now it bites us. need to find a mitigation).

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/28901/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-10-6-11-48-00.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-10-6-11-48-34.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/file-alice-2022-10-6-11-42-14.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/folder-alice-2022-10-6-11-41-00.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/folder-brian-2022-10-6-11-41-07.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/unstructured-collection-of-testable-space-interactions-alice-2022-10-6-11-44-49.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/unstructured-collection-of-testable-space-interactions-anonymous-2022-10-6-11-45-41.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/28901/tracing/unstructured-collection-of-testable-space-interactions-brian-2022-10-6-11-45-43.zip 


</p></details>
:boom: The e2e-ocis tests pipeline failed. The build has been cancelled.

File: packages/web-client/src/webdav/moveFiles.ts [hunk 1]
```diff
@@ -9,8 +14,15 @@ export const MoveFilesFactory = ({ sdk }: WebDavOptions) => {
       { path: sourcePath },
       targetSpace: SpaceResource,
       { path: targetPath },
-      options?: { overwrite?: boolean }
+      options?: { overwrite?: boolean; shareJailId?: string }
```
Comment (kulmann) [inline packages/web-client/src/webdav/moveFiles.ts:17]: @JanAckermann please remove the shareJailId option, not used but imported as discussed.

**Comment by JammingBen** [comment]
Issues I've found:

* Share a text file via public link -> Open link -> Text editor does not open the contents of the file
* Share a PDF file via public link -> Open link -> PDF viewer does not open the contents of the file
* Breadcrumb is broken when having more than 2 levels

**Comment by JanAckermann** [comment]
Issues I've found:

Clicking in on rename action in the context actions besides the breadcrumbs on a received share will show the original share name despite the fact, the share receiver has renamed the share. 
This seems have to do something with the unified folder loader and isn't the case on master.

**Comment by kulmann** [comment]
> * Breadcrumb is broken when having more than 2 levels

Mitigated in a recent commit by omitting the fileId of the current folder from the target route query (it was there because we copy the query of the current route to the target route for good reasons).
Obviously this disables navigation by fileId for breadcrumbs on parent folders. For that we'll need propfinds on each parent. Out of scope for this PR.

**Comment by dschmidt** [comment]
Sharing a single file via public link is broken because of https://github.com/owncloud/ocis/issues/4758

File: packages/web-client/src/webdav/listFiles.ts [hunk 2]
```diff
@@ -62,6 +62,7 @@ export const ListFilesFactory = ({ sdk }: WebDavOptions) => {
         )
         const resources = webDavResources.map(buildResource)
         if (fileId && fileId !== resources[0].fileId) {
+          console.log('nono', path, fileId, resources[0].fileId)
```
Comment (dschmidt) [inline packages/web-client/src/webdav/listFiles.ts:65]: 🙈

File: packages/web-pkg/src/composables/appDefaults/useAppFolderHandling.ts [hunk 3]
```diff
@@ -51,7 +51,9 @@ export function useAppFolderHandling({
       context = unref(context)
       const space = unref(context.space)
 
-      const pathResource = await getFileInfo(context, { davProperties: [DavProperty.Name] })
+      const pathResource = await getFileInfo(context, {
+        davProperties: [DavProperty.FileId]
+      })
```
Comment (dschmidt) [inline packages/web-pkg/src/composables/appDefaults/useAppFolderHandling.ts:56]: Don't we need DavProperty.Name for the path?!

**Comment by JammingBen** [comment]
It also fixes https://github.com/owncloud/web/issues/7715 and https://github.com/owncloud/web/issues/7714, right?

File: packages/web-app-draw-io/src/App.vue [hunk 4]
```diff
@@ -125,6 +125,7 @@ export default defineComponent({
       }
     },
     load() {
+      this.checkPermissions()
```
Comment (lookacat) [inline packages/web-app-draw-io/src/App.vue:128]: load isnt async and await is missing

File: packages/web-pkg/src/composables/appDefaults/useAppFolderHandling.ts (hunk 3 above)
Comment (kulmann) [reply packages/web-pkg/src/composables/appDefaults/useAppFolderHandling.ts:56]: @dschmidt I was wrong to say that we need the `name` prop. `name` is only the display name of the file.

**Comment by kulmann** [comment]
> It also fixes #7715 and #7714, right?

yep

File: packages/web-app-draw-io/src/App.vue [hunk 5]
```diff
@@ -125,6 +125,7 @@ export default defineComponent({
...
       this.getFileContents(this.currentFileContext)
         .then((resp) => {
```
Comment (lookacat) [inline packages/web-app-draw-io/src/App.vue:130]: then can be replaced with async await

File: packages/web-app-draw-io/src/App.vue (hunk 4 above)
Comment (kulmann) [reply packages/web-app-draw-io/src/App.vue:128]: Not needed. `checkPermissions` and `getFileContents` can run in parallel. No need to wait until `checkPermissions` is finished.

Comment (kulmann) [reply packages/web-app-draw-io/src/App.vue]: Pushed a commit to make that easier to see

**Review by JammingBen — APPROVED** [review]

**Pull request merged**